validator/keymanager/v2/derived/derived.go

```diff
@@ -406,27 +402,38 @@ func (dr *Keymanager) DepositDataForAccount(accountIndex uint64) ([]byte, error)
 	return tx.Data(), nil
 }
 
-func (dr *Keymanager) initializeSecretKeysCache() error {
-	dr.lock.Lock()
-	defer dr.lock.Unlock()
-	for i := uint64(0); i < dr.seedCfg.NextAccount; i++ {
+// Append public and secret keys for a given account index to their respective caches
+func (dr *Keymanager) appendKeyesToCaches(secretKey bls.SecretKey) error {
```

Comment:
No mutex here? Would be easy to abuse this function since it is not thread safe. Can we move the mutex locking inside if possible?

Reply:
if this is not reasonable, we should at least add a warning or comment that this function isn't thread safe

Reply:
Sorry about all the typos. i need to install a spellchecker into my VS Code

Reply:
If we remove the mutex lock around initializeKeysCaches then we can place the mutex lock inside the appendKeyesToCaches, if we don't rremove the lock around or inside initializeKeysCaches , we would get a deadlock because initializeKeysCaches is calling appendKeyesToCaches.

Reply:
Should i go ahead and remove the lock around initializeKeysCaches and put a lock inside appendKeyesToCaches?